### PR TITLE
Drop support for Node 12

### LINKED ISSUE #328: Drop Node 12

### PR DESCRIPTION
**Pull request by chriskrycho**
- update CI config to use Node 14
- remove Node 12 from engines, add Node 16
- update to latest Node 14 with Volta
- update compatibility note in README

Resolves #328

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
     "webpack": "^5.59.1"
   },
   "engines": {
-    "node": "12.* || >= 14.*"
+    "node": "14.* || >= 16.*"
```
Comment (SergeAstapov) [inline package.json:82]: @chriskrycho should we actually do `14.* || 16.* || >= 18.*`?

Comment (chriskrycho) [reply package.json:82]: Ah, yep.

Comment (chriskrycho) [reply package.json]: Done!

**Comment by chriskrycho** [comment]
Good call – done!

**Pull request merged**